src/main/scala/fs/fusejna.scala

```diff
@@ -0,0 +1,114 @@
+package suffuse
+package fs
+
+import net.fusejna._
+import jio._
+import scala.util.Properties.isMac
+
+/** Cleaning up the javacentric fuse-jna types.
+ */
+
+object Node {
+  import net.fusejna.types.TypeMode.NodeType
+
+  final val BlockDev = NodeType.BLOCK_DEVICE
+  final val Dir      = NodeType.DIRECTORY
+  final val Fifo     = NodeType.FIFO
+  final val File     = NodeType.FILE
+  final val Link     = NodeType.SYMBOLIC_LINK
+  final val Socket   = NodeType.SOCKET
+}
+
+/** Widening access so we don't have to use inheritance everywhere.
+ */
+
+trait FuseFs extends FuseFilesystem {
+  def unmountTry(): Unit = (
+    if (!isMounted)
+      return
+    else if (isMac)
+      exec("umount", "-f", getMountPoint.getPath) orElse exec("diskutil", "unmount", getMountPoint.getPath)
+    else
+      exec("fusermount", "-u", getMountPoint.getPath)
+  )
+  private def doMount(mountPoint: File, blocking: Boolean): this.type = {
+    addUnmountHook(this)
+    super.mount(mountPoint, blocking)
+    this
+  }
+
+  def mountBg(mountPoint: Path): this.type = doMount(mountPoint.toFile, false)
+  def mountFg(mountPoint: Path): this.type = doMount(mountPoint.toFile, true)
+
+  def fuseContext()         = super.getFuseContext
+  override def getOptions() = fs.defaultOptions
+}
+
+/** This makes it easy to modify or extends the behavior of an existing
+ *  filesystem instance by overriding a small handful of methods.
+ */
+abstract class ForwarderFs extends FuseFs {
```

Comment:
Oh, I totally agree. First have (quite) a few examples lying around, otherwise you keep shuffling with the abstractions. It's one of the first lessons I teach my colleagues :-)